### ISSUE TITLE
no-duplicate-variable rule false positives

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#372]
Examples:

duplicate "key":
```ts
export interface IClipboard {
    copy(key: string, state: any): void;
    paste(key: string): any;
}
```

duplicate "values":
```ts
findMaxOrMin(values: any[], defaultValue: number, operation: (...values: any[]) => number)
```

duplicate "e":
```ts
try {
    //
} catch (e) {
    e.blah();
    //
}

try {
    //
} catch (e) {
    e.blah();
    //
}
```
